src/subscription.js

```diff
@@ -0,0 +1,278 @@
+            leagueNames.push(l.options.name);
+        });
+        return "You didn't specify a valid league. These are your options:\n" +
+            "```" + leagueNames.join("\n") + "```";
+    });
+}
+
+//------------------------------------------------------------------------------
+// Tell them they can only listen for themselves
+//------------------------------------------------------------------------------
+function formatCanOnlyListenForYouResponse(config, listener) {
+    return Q.fcall(function() {
+        return "You can't subscribe " + listener + " because you aren't a moderator." +
+            " As a non-moderator you can only subscribe yourself. Use 'me' as " + 
+            " listener.";
+    });
+}
+
+//------------------------------------------------------------------------------
+// This is not a listener that we recognize
+//------------------------------------------------------------------------------
+function formatInvalidListenerResponse(config, listener) {
+    return Q.fcall(function() {
+        return "I don't recognize '" + listener + "'. Please use 'me' as the listener.";
+    });
+}
+
+//------------------------------------------------------------------------------
+// Format the invalid event handler
+//------------------------------------------------------------------------------
+function formatInvalidEventResponse(config, event) {
+    return Q.fcall(function() {
+        return "`" + event + "` is not one of the events that I recognize. Use one of: " +
+            "```" + events.join("\n") + "```";
+    });
+}
+
+//------------------------------------------------------------------------------
+// Format the response for when the source is not valid
+//------------------------------------------------------------------------------
+function formatInvalidSourceResponse(config, source) {
+    return Q.fcall(function() {
+        return "`" + source + "` is not a valid source of that event. Please target a valid user.";
+    });
+}
+
+//------------------------------------------------------------------------------
+// Processes a tell command.
+//
+// A tell command is made up like so:
+//      tell <listener> when <event> in <league> for <source>
+//------------------------------------------------------------------------------
+function processTellCommand(config, message) {
+    return Q.fcall(function() {
+        var requester = player.getSlackUserFromNameOrID(slack.users, message.user);
+        var components = message.text.split(" ");
+        var args = components.slice(0, 7);
+        var sourceName = components.splice(7).join(" ");
+        var tell = args[0],
+            listener = args[1],
+            when = args[2],
+            event = args[3],
+            _in = args[4], 
+            targetLeague = args[5], 
+            _for = args[6];
+
+        // Ensure the basic command format is valid
+        if (!_.isEqual(["tell", "when", "in", "for"], [tell, when, _in, _for])) {
```

Comment:
In python it's even more elegant:
 tell, listener, when, event, _in, targetLeague, _for = message.text.split(" ", 7)